### PR TITLE
Docs, part 15

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #358

### CODE REVIEW

**Review by vjik — APPROVED** [review]

File: src/Rule/Composite.php [hunk 1]
```diff
@@ -37,27 +83,41 @@ class Composite implements
     use WhenTrait;
 
     /**
-     * @var iterable<int, RuleInterface>
+     * @var iterable<int, RuleInterface> A set of normalized rules that needs to be grouped.
```
Comment (samdark) [inline src/Rule/Composite.php:86]: Need to separate annotation to psalm-var since it isn't standard phpdoc.

File: src/Rule/Each.php [hunk 2]
```diff
@@ -22,7 +23,38 @@
 use Yiisoft\Validator\WhenInterface;
 
 /**
- * Validates an array by checking each of its elements against a set of rules.
+ * Allows to define a set of rules for validating each element of an iterable.
+ *
+ * An example for simple iterable that can be used to validate RGB point:
```
Comment (samdark) [inline src/Rule/Each.php:28]:
```suggestion
 * An example for simple iterable that can be used to validate RGB color:
```

File: src/Rule/StopOnError.php [hunk 3]
```diff
@@ -19,7 +19,22 @@
 use Yiisoft\Validator\WhenInterface;
 
 /**
- * Can be used for validation of nested structures.
+ * A higher-order rule, applies to a set of rules, runs validation for each one of them in the order they are defined
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/Rule/StopOnError.php:22]: Similar to higher order functions.

[the remark this answers is not in the document]
Comment (samdark) [reply src/Rule/StopOnError.php:22]: I'd avoid functional programming terminology. I doubt it's known enough among readers of this doc.

[the remark this answers is not in the document]
Comment (arogachev) [reply src/Rule/StopOnError.php:22]: Discussed it and decided to remove as ambiguous.

**Pull request merged**